src/client/components/DicePanel.tsx

```diff
@@ -1,57 +1,382 @@
+      ],
+      optimisticKey: "diceTemplateCategories",
+      syncToServerThrottle: DEFAULT_SYNC_TO_SERVER_DEBOUNCE_TIME,
+    });
+  }
+
+  const d20Part: RRDiceTemplatePartDice = {
+    id: rrid<RRDiceTemplatePart>(),
+    type: "dice",
+    faces: 20,
+    count: 1,
+    negated: false,
+    modified: "none",
+    damage: { type: null },
+  };
+  const characters = useServerState((state) => state.characters);
+  const character = characters.entities[myself.mainCharacterId!];
+
+  function getSavingThrowTemplates() {
+    const templates = [];
+
+    templates.push(
+      ...characterStatNames.map(
+        (statName: typeof characterStatNames[number]) => {
+          const proficiency =
+            character === undefined ? 0 : character.savingThrows[statName] ?? 0;
+
+          const parts: RRDiceTemplatePart[] = [
+            d20Part,
+            {
+              id: rrid<RRDiceTemplatePart>(),
+              type: "linkedStat",
+              name: statName,
+              damage: { type: null },
+            },
+          ];
+
+          if (proficiency !== 0)
+            parts.push({
+              id: rrid<RRDiceTemplatePart>(),
+              type: "linkedProficiency",
+              damage: { type: null },
+              proficiency,
+            });
+
+          return {
+            id: rrid<RRDiceTemplate>(),
+            name: `${statName} Saving Throw`,
+            notes: "",
+            parts,
+            rollType: null,
+          };
+        }
+      )
+    );
+    return templates;
+  }
+
+  function getSkillTemplates() {
+    const templates: RRDiceTemplate[] = [];
+
+    templates.push(
+      ...skillNames.map((skill) => {
+        const proficiency =
+          character === undefined ? 0 : character.skills[skill] ?? 0;
+        const parts: RRDiceTemplatePart[] = [
+          d20Part,
+          {
+            id: rrid<RRDiceTemplatePart>(),
+            type: "linkedStat",
+            name: skillMap[skill],
+            damage: { type: null },
+          },
+        ];
+        if (proficiency !== 0)
+          parts.push({
+            id: rrid<RRDiceTemplatePart>(),
+            type: "linkedProficiency",
+            damage: { type: null },
+            proficiency,
+          });
+
+        return {
+          id: rrid<RRDiceTemplate>(),
+          name: skill,
+          notes: "",
+          parts,
+          rollType: null,
+        };
+      })
+    );
+    return templates;
+  }
+
+  function renderContent(active: string) {
+    const category = myself.diceTemplateCategories.find(
+      (cat) => cat.id === active
+    );
+    if (active !== "Skills" && active !== "STs" && category === undefined)
+      active = "Input";
+    switch (active) {
+      case "Input":
+        return (
+          <>
+            <DiceInterface />
+            <DiceInput />
+          </>
+        );
+      case "Skills":
+        return (
+          <GeneratedDiceTemplates
+            templates={getSkillTemplates()}
```

Comment:
Can we `useMemo` the result of `getSkillTemplates()` so that the generated dice templates are only re-calculated if the character proficiencies change?

I.e. somewhere above:
```ts
const skillTemplates = useMemo(
	() => skillNames.map((skill) => { /* ... */ }),
	[character.proficiencies]
);
```